### PR TITLE
fix(instrumentation): replace api base url hardcoded value from watsonx

### LINKED ISSUE #2163: 🐛 Bug Report: Watsonx hardcoded values
### Which component is this bug for?

Watsonx Instrumentation

### 📜 Description

Some variables, showing in traceloop traces, are hardcoded.
For example, even if hitting watsonx in frankfurt, it shows that it's using https://us-south.ml.cloud.ibm.com, which is incorrect.

### 👟 Reproduction steps

Create a python script using watsonx in frankfurt for example, and see that in the traces it shows https://us-south.ml.cloud.ibm.com, as hardcoded in packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py line 103

### 👍 Expected behavior

It should show the real parameter istead of harcoded ones.

### 👎 Actual Behavior with Screenshots

image

### 🤖 Python Version

3.12

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by rajmanna-dev**
Fixed #2163 

I am a beginner in learning python, I don't know how to make or add test for this changes. I need guidence.

- [ ] I have added tests that cover my changes.
- [X] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.
- [X] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [X] (If applicable) I have updated the documentation accordingly.

### CODE REVIEW

File: packages/sample-app/sample_app/watsonx-langchain.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def watsonx_llm_init() -> ModelInference:
 
     watsonx_llm = WatsonxLLM(
         model_id="ibm/granite-13b-instruct-v2",
-        url="https://us-south.ml.cloud.ibm.com",
+        url=os.getenv("API_BASE_URL"),
```
Comment (nirga) [inline packages/sample-app/sample_app/watsonx-langchain.py:31]: Can you keep the original code?

**Review by nirga — COMMENTED** [review]
Thanks @rajmanna-dev! Left a small comment. Can you sign the CLA?

File: packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 from opentelemetry.trace.status import Status, StatusCode
 from opentelemetry.metrics import get_meter
 from opentelemetry.metrics import Counter, Histogram
+from sample-app.sample_app.watsonx-langchain import watsonx_llm_init
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:17]: remove?

**Review by nirga — COMMENTED** [review]
@rajmanna-dev note that CI is failing - I think it's related to the import you added

File: packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py [hunk 3]
```diff
@@ -101,10 +102,15 @@ def _set_span_attribute(span, name, value):
 
 
 def _set_api_attributes(span):
+    watsonx_llm = watsonx_llm_init()
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:105]: why do you need this?

File: packages/sample-app/sample_app/watsonx-langchain.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ def watsonx_llm_init() -> ModelInference:
 
     watsonx_llm = WatsonxLLM(
         model_id="ibm/granite-13b-instruct-v2",
-        url="https://us-south.ml.cloud.ibm.com",
+        url=os.getenv("API_BASE_URL", "https://us-south.ml.cloud.ibm.com"),
```
Comment (nirga) [inline packages/sample-app/sample_app/watsonx-langchain.py:31]: remove?

File: packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py (hunk 3 above)
Comment (rajmanna-dev) [reply packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:105]: How can i extract `url` direct from WatsonxLLM objcet.

Comment (nirga) [reply packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:105]: See the comment here I shared an example of how we've done it elsewhere - https://github.com/traceloop/openllmetry/pull/2165#pullrequestreview-2374362135

Comment (rajmanna-dev) [reply packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:105]: Should i check the attribute 'url' of the instance and then set that to api_base_url variable.
Like:

```python
if hasattr(instance, "url"):
    api_base_url = instance.url
```

Comment (nirga) [reply packages/opentelemetry-instrumentation-watsonx/opentelemetry/instrumentation/watsonx/__init__.py:105]: Have you checked that the instance can actually have this attribute? I'd run one of the sample apps and see

**Pull request closed without merging**